src/pages/Environments.js

```diff
@@ -145,75 +305,67 @@ const Environments = () => {
+            cellRenderer: ({ rowIndex }) => {
+              const cloudEnvironment = filteredCloudEnvironments[rowIndex]
+              // TODO: update return logic once we support more app types (will need a backend change to return appType in list apps endpoint as well)
+              return cloudEnvironment.appName ? 'Galaxy' : (cloudEnvironment.runtimeConfig.cloudService === 'DATAPROC' ? 'Dataproc' : cloudEnvironment.runtimeConfig.cloudService)
             }
           },
           {
             size: { basis: 90, grow: 0 },
             headerRenderer: () => 'Details',
             cellRenderer: ({ rowIndex }) => {
-              const { runtimeName, runtimeConfig: { persistentDiskId } } = filteredRuntimes[rowIndex]
-              const disk = _.find({ id: persistentDiskId }, disks)
-              return h(PopupTrigger, {
-                content: div({ style: { padding: '0.5rem' } }, [
-                  div([span({ style: { fontWeight: '600' } }, ['Name: ']), runtimeName]),
-                  disk && div([span({ style: { fontWeight: '600' } }, ['Persistent Disk: ']), disk.name])
-                ])
-              }, [h(Link, ['details'])])
+              const cloudEnvironment = filteredCloudEnvironments[rowIndex]
+              return cloudEnvironment.appName ? renderDetailsApp(cloudEnvironment, disks) : renderDetailsRuntime(cloudEnvironment, disks)
             }
           },
           {
             size: { basis: 150, grow: 0 },
             field: 'status',
             headerRenderer: () => h(Sortable, { sort, field: 'status', onSort: setSort }, ['Status']),
             cellRenderer: ({ rowIndex }) => {
-              const runtime = filteredRuntimes[rowIndex]
-              return h(Fragment, [
-                runtime.status,
-                runtime.status === 'Error' && h(Clickable, {
-                  tooltip: 'View error',
-                  'aria-label': 'View error',
-                  onClick: () => setErrorRuntimeId(runtime.id)
-                }, [icon('warning-standard', { style: { marginLeft: '0.25rem', color: colors.danger() } })])
-              ])
+              const cloudEnvironment = filteredCloudEnvironments[rowIndex]
+              return cloudEnvironment.appName ? renderErrorApps(cloudEnvironment) : renderErrorRuntimes(cloudEnvironment)
             }
           },
           {
             size: { basis: 120, grow: 0.2 },
             headerRenderer: () => 'Location',
             cellRenderer: ({ rowIndex }) => {
-              const { runtimeConfig: { zone, region } } = filteredRuntimes[rowIndex]
-              return zone || region
+              const cloudEnvironment = filteredCloudEnvironments[rowIndex]
+              // This logic works under the assumption that all Galaxy apps get created in zone 'us-central1-a'
+              const zone = cloudEnvironment.runtimeConfig ? cloudEnvironment.runtimeConfig.zone : 'us-central1-a'
```

Comment:
the reason I did it this way was because for GCE runtimes we have access to `zone` whereas for for Dataproc we have access to `region` - i can try to think of a way to clean up this logic. we display that `zone` or `region` as the "Location" column

Reply:
Right. I think what Mohs was pointing out is…the default below for `region` (`us-central1-a`) is actually a `zone` value. In other words it should be `us-central1` instead.

Also, optional chaining could simplify the two definitions, like so:

```javascript
const zone = cloudEnvironment?.runtimeConfig?.zone || 'us-central1-a'
const region = cloudEnvironment?.runtimeConfig?.region || 'us-central1'
```

and then we could just

```javascript
return zone || region
```

Reply:
ah i see, will update!

Reply:
I did
```
const zone = cloudEnvironment?.runtimeConfig?.zone
const region = cloudEnvironment?.runtimeConfig?.region
return zone || region || 'us-central1-a'
```
instead because we don't want to return `'us-central1-a'` for Dataproc